cpp/marc_tools/convert_marc_to_utf8.cc

```diff
@@ -0,0 +1,97 @@
+/** \brief Utility for patching up incorrect encodings in MARC files o/w assumed to be UTF8.
+ *  \author Dr. Johannes Ruscheinski (johannes.ruscheinski@uni-tuebingen.de)
+ *
+ *  \copyright 2022 Universitätsbibliothek Tübingen.  All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <cstdlib>
+#include "MARC.h"
+#include "TextUtil.h"
+#include "util.h"
+
+
+namespace {
+
+
+[[noreturn]] void Usage() {
+    ::Usage("(--all-fields|--non-utf8-fields-only) from_encoding marc_input marc_output\n"
```

Comment:
can we also add a parameter `to_encoding` instead of hardcoding `utf8` everywhere? As far as i can see, `EncodingConverter` already supports `to_encoding` anyway. Of course this means that `--non-utf8-fieds-only` needs to be renamed too, as well as the script name being changed to something more generic.

Reply:
No because the MARC-21 standard only supports two encodings, UTF8 and the proprietary MARC-8.